### PR TITLE
Add index on STI type columns

### PR DESCRIPTION
**Pull request by juliancheal**
This follows on from pervious work in this PR https://github.com/ManageIQ/manageiq/pull/9516

I created this as a new migration and not a cherry-pick to give us an updated timestamp.

/cc @Fryguy @kbrock

### CODE REVIEW

**Comment by juliancheal** [comment]
@miq-bot add_label performance "sql migration" wip

**Comment by juliancheal** [comment]
@miq-bot assign @Fryguy

**Comment by juliancheal** [comment]
Looks like there are a few tables that don't exist anymore. Will update PR.

**Comment by juliancheal** [comment]
@kbrock @Fryguy thoughts?

**Comment by Fryguy** [comment]
Seems ok to me.  @kbrock Thoughts?

**Comment by kbrock** [comment]
I had thought we may want to make compound indexes out of these instead of just `type`.
But think merging it should be good enough

**Comment by juliancheal** [comment]
@kbrock what's a compound index?

**Comment by juliancheal** [comment]
Ping @Fryguy @kbrock

File: db/migrate/20180425123859_add_index_on_all_type_columns.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    add_index :container_groups, :type
+    add_index :container_nodes, :type
+    add_index :container_volumes, :type
+    add_index :containers, :type
+    add_index :customization_scripts, :type
+    add_index :customization_templates, :type
+    add_index :dialog_fields, :type
+    add_index :ems_clusters, :type
+    add_index :ems_folders, :type
+    add_index :event_streams, :type
```
Comment (kbrock) [inline db/migrate/20180425123859_add_index_on_all_type_columns.rb]: Lets not add an index to `event_streams` unless we know it is needed for an index.
This table is BIG

Comment (juliancheal) [reply db/migrate/20180425123859_add_index_on_all_type_columns.rb]: Removed `event_streams`

**Comment by juliancheal** [comment]
@Fryguy merge on 🍏 ?

**Pull request merged**